src/Term.hs

```diff
@@ -22,6 +22,7 @@ data Lit
 
 data Type
   = TInt
+  | TVar Name
   | TBool
   | TString
   | TClosure Type Type -- A closure
```

Comment:
Not gonna lie, the comment seems useless here.
`-- A closure with arguments and a return type`
seemed more appropriate.

Reply:
Good first comment.
/joke
I am also going to add this and remove the parser part. Basically making another EDSL in Haskell. To experiment substitutions.